app/api/__tests__/insert-dturn-statement.js

```diff
@@ -0,0 +1,117 @@
+// https://github.com/EnCiv/civil-pursuit/issues/163
+
+import insertDturnStatement from '../insert-dturn-statement'
+import { initDiscussion } from '../../dturn/dturn'
+
+import { Mongo } from '@enciv/mongo-collections'
+import { MongoMemoryServer } from 'mongodb-memory-server'
+import { MongoClient, ObjectId } from 'mongodb'
+
+const { BSON } = require('bson')
+
+const Points = require('../../models/points')
+
+// Config
+
+const nonExistentDiscussionId = new ObjectId()
+const existentDiscussionId = new ObjectId()
+const synuser = { synuser: { id: '6667d5a33da5d19ddc304a6b' } }
+
+const pointObjNoId = { title: 'NoIdTitle', description: 'NoIdDesc' }
+const pointObjWithId = {
+  _id: new ObjectId('6667d688b20d8e339ca50020'),
+  title: 'WithIdTitle',
+  description: 'WithIdDesc',
+}
+
+let MemoryServer
+
+beforeEach(async () => {
+  jest.spyOn(console, 'error').mockImplementation(() => {})
+})
+
+afterEach(async () => {
+  console.error.mockRestore()
+})
+
+beforeAll(async () => {
+  MemoryServer = await MongoMemoryServer.create()
+  const uri = MemoryServer.getUri()
+  await Mongo.connect(uri)
+
+  await initDiscussion(existentDiscussionId, obj => {})
```

Comment:
initDiscussion's second prop in an object, called options. 

can you do something like:
```

const UInfoHistory=[]

initDiscussion(id, { updateUinfo: obj => UInfoHistory.push(obj) }

```

And then in the tests, test the object in UInfoHistory - just make sure it's the same every time.

It will need a fixed ObjectId for existingDiscussionId.

You can use ../enciv-home/node_modules/.bin/mongo-id to generate a mongo id string.  If you have enciv-home installed. 

In the api that setups the subscription to the discussion, we will pass an unpdateUInfo function that saves those to the database.  And when the discussion is loaded, the getAllUInfo option will fetch all the UInfo records from the db to restore the previous state of the discussion.

Reply:
I'm having a bit of difficulty working the updateUInfo function, not sure if I'm missing something obvious - but whenever I check UInfoHistory it's empty.